server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -194,14 +194,27 @@ public IndexNameExpressionResolver(Settings settings) {
             }
 
             Collection<IndexMetaData> resolvedIndices = aliasOrIndex.getIndices();
-            if (resolvedIndices.size() > 1 && !options.allowAliasesToMultipleIndices()) {
-                String[] indexNames = new String[resolvedIndices.size()];
-                int i = 0;
-                for (IndexMetaData indexMetaData : resolvedIndices) {
-                    indexNames[i++] = indexMetaData.getIndex().getName();
+            if (resolvedIndices.size() > 1 && options.allowAliasesToMultipleIndices() && options.requireAliasesToWriteIndex()) {
+                // find write-index
+                if (aliasOrIndex.isAlias()) {
+                    AliasOrIndex.Alias alias = (AliasOrIndex.Alias)  aliasOrIndex;
+                    List<IndexMetaData> writeIndices = alias.getWriteIndices();
+                    if (writeIndices.size() == 1) {
+                        resolvedIndices = writeIndices;
+                    } else if (writeIndices.size() > 1) {
+                        String[] indexNames = new String[writeIndices.size()];
+                        int i = 0;
+                        for (IndexMetaData indexMetaData : writeIndices) {
+                            indexNames[i++] = indexMetaData.getIndex().getName();
+                        }
+                        throw new IllegalArgumentException("Alias [" + expression +
+                            "] has multiple write indices " + Arrays.toString(indexNames)
+                            + " associated with it with [is_write_index = true]");
```

Comment:
is this something that can happen in reality? I was under the impression that we would try and make sure that an alias that points to multiple indices has a single write index. Feels a bit weird to have to check for this every time we resolve an index expression.

Reply:
We do not upgrade existing alias configurations, or auto-set `is_write_index` for alias update api calls. So indices may remain with no write indices

Reply:
but here we are validating that an alias does not have multiple write indices rather than no write indices?

Reply:
oh, I see it now. sorry. did not expand for context. You're right. This is a leftover exception from a previous way of doing things. I will clean this up. thanks!